framework/sysdeps/windows/child_debug.cpp

```diff
@@ -5,15 +5,59 @@
 
 #include "sandstone.h"
 #include "sandstone_p.h"
+#include "sandstone_context_dump.h"
+
+#include <windows.h>
+
+static LONG WINAPI handler(EXCEPTION_POINTERS *info)
+{
+    // send the entire context - find out what XState feature is the furthest out
+    ptrdiff_t xsave_size = sizeof(*info->ContextRecord);
+    const auto start = reinterpret_cast<char *>(info->ContextRecord);
+#ifdef __x86_64__
+    if (DWORD64 xsave_bv; GetXStateFeaturesMask(info->ContextRecord, &xsave_bv) && xsave_bv) {
+        // these are already in the fixed context record:
+        xsave_bv &= ~XSTATE_MASK_LEGACY;
+        for (int bit = XSTATE_AVX; xsave_bv; ++bit) {
+            DWORD length;
+            DWORD64 mask = DWORD64(1) << bit;
+            if ((xsave_bv & mask) == 0)
+                continue;
+            xsave_bv &= ~mask;
+            void *where = LocateXStateFeature(info->ContextRecord, bit, &length);
+            if (where)
+                xsave_size = std::max(xsave_size, static_cast<char *>(where) + length - start);
+        }
+    }
+#endif
+
+    WriteFile(HANDLE(sApp->shmem->debug_mailslot), start, xsave_size, nullptr, nullptr);
+
+    return EXCEPTION_CONTINUE_SEARCH;
+}
 
 void debug_init_child()
 {
+    if (!SandstoneConfig::ChildDebugCrashes || sApp->shmem->debug_mailslot == -1)
+        return;
+
+    // install the vectored exception handler
+    PVOID h = AddVectoredExceptionHandler(true, handler);
+    (void) h;
 }
 
 void debug_init_global(const char *on_hang_arg, const char *on_crash_arg)
 {
     (void) on_hang_arg;
-    (void) on_crash_arg;
+    if (SandstoneConfig::ChildDebugCrashes) {
+        std::string_view arg = on_crash_arg;
+        if (arg == "context") {
+            sApp->shmem->debug_mailslot = open_mailslot(MemfdInheritOnExec);
+        } else if (arg != "kill") {
```

Comment:
We're going to need to either update bats along with this change or handle other `on-crash` actions here.

Reply:
Fixed.